tapgarden/planter.go

```diff
@@ -669,9 +670,95 @@ func (c *ChainPlanter) newBatch() (*MintingBatch, error) {
 	return newBatch, nil
 }
 
+// preCommitmentOutput creates the pre-commitment output for a batch that uses
+// universe commitments.
+func preCommitmentOutput(pendingBatch *MintingBatch) (*wire.TxOut,
+	*btcec.PublicKey, *btcec.PublicKey, error) {
```

Comment:
We could add the `wire.TxOut` of the output to the `PreCommitmentOutput` struct and return that here. Then we can have a single option in `fundGenesisPsbt` and do the validation in a `Validate()` method to reduce the size of that already long function a bit.

Reply:
Elsewhere, `PreCommitmentOutput` is populated from the database where the tx ouput is missing. We only have the index at that point.

Reply:
Yes, I saw that. But can't we re-create the tx output from the database fields? Then that could actually even be a method on the struct that generates the `txOut`.

Reply:
I've revised `preCommitmentOutput` along the lines you suggested. Please let me know if that's what you had in mind.

Reply:
Yeah, this is exactly what I had in mind. Nice!